src/tools/genpolicy/src/registry.rs

```diff
@@ -63,57 +71,34 @@ pub struct ImageLayer {
 impl Container {
     pub async fn new(use_cached_files: bool, image: &str) -> Result<Self> {
         info!("============================================");
-        info!("Pulling manifest and config for {:?}", image);
-        let reference: Reference = image.to_string().parse().unwrap();
-        let auth = build_auth(&reference);
-
-        let mut client = Client::new(ClientConfig {
-            platform_resolver: Some(Box::new(linux_amd64_resolver)),
-            ..Default::default()
-        });
-
-        match client.pull_manifest_and_config(&reference, &auth).await {
-            Ok((manifest, digest_hash, config_layer_str)) => {
-                debug!("digest_hash: {:?}", digest_hash);
-                debug!(
-                    "manifest: {}",
-                    serde_json::to_string_pretty(&manifest).unwrap()
-                );
-
-                // Log the contents of the config layer.
-                if log::max_level() >= LevelFilter::Debug {
-                    let mut deserializer = serde_json::Deserializer::from_str(&config_layer_str);
-                    let mut serializer = serde_json::Serializer::pretty(io::stderr());
-                    serde_transcode::transcode(&mut deserializer, &mut serializer).unwrap();
-                }
-
-                let config_layer: DockerConfigLayer =
-                    serde_json::from_str(&config_layer_str).unwrap();
-                let image_layers = get_image_layers(
-                    use_cached_files,
-                    &mut client,
-                    &reference,
-                    &manifest,
-                    &config_layer,
-                )
-                .await
-                .unwrap();
+        info!("Pulling image and layers for {:?}", image);
 
-                Ok(Container {
-                    config_layer,
-                    image_layers,
-                })
-            }
-            Err(oci_distribution::errors::OciDistributionError::AuthenticationFailure(message)) => {
-                panic!("Container image registry authentication failure ({}). Are docker credentials set-up for current user?", &message);
-            }
+        let containerd_socket_path = "/var/run/containerd/containerd.sock";
+        let client = match containerd_client::Client::from_path("/var/run/containerd/containerd.sock").await {
```

Comment:
Looks like a duplicate definition of the same path. a static constant declared at the beginning of registry.rs should work here.